test/Microsoft.AspNetCore.WebSockets.Server.Test/AutobahnTests.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.Testing;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.AspNetCore.WebSockets.Server.Test.Autobahn;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.PlatformAbstractions;
+using Xunit;
+
+namespace Microsoft.AspNetCore.WebSockets.Server.Test
+{
+    public class AutobahnTests
+    {
+        // Skip if wstest is not installed for now, see https://github.com/aspnet/WebSockets/issues/95
+        // We will enable Wstest on every build once we've gotten the necessary infrastructure sorted out :).
+        [ConditionalFact]
+        [SkipIfWsTestNotPresent]
+        public async Task AutobahnTestSuite()
+        {
+            var outDir = Path.Combine(PlatformServices.Default.Application.ApplicationBasePath, "autobahnreports").Replace("\\", "\\\\");
+            if (Directory.Exists(outDir))
+            {
+                Directory.Delete(outDir, recursive: true);
+            }
+
+            var spec = new AutobahnSpec(outDir)
+                .IncludeCase("*")
+                .ExcludeCase("12.*", "13.*");
+            var loggerFactory = new LoggerFactory().AddConsole();
+
+            AutobahnResult result;
+            using (var tester = new AutobahnTester(loggerFactory, spec))
+            {
+                await tester.DeployTestAndAddToSpec(ServerType.Kestrel, ssl: false, environment: "ManagedSockets", expectationConfig: expect => expect
```

Comment:
I started with that, but the problem is that completely borks up the Autobahn reporting. The test suite is designed to run multiple Server test cases and produce a single report aggregating all that. We could do it, but we'd have 4+ different reports rather than one HTML report (which I intend to publish as an artifact)